lms/static/scripts/frontend_apps/utils/fetch.ts

```diff
@@ -51,6 +53,10 @@ export function useFetch<T = unknown>(
   key: string | null,
   fetcher?: Fetcher<T>,
 ): FetchResult<T> {
+  // If an InitialLoadingContext is found, we want to notify about loading
+  // start/finish
+  const globalLoading = useContext(InitialLoadingContext);
```

Comment:
Applying this globally in `useFetch` means that we'll apply this for all fetches, even those for auxiliary / less important content.

Reply:
Yep, that's true. Would you add an option to `useFetch` to conditionally handle this, or would you leave `useFetch` as is and add a side effect to individual components where we want to integrate with the global loading?